lib/autocomplete_source.rb

```diff
@@ -11,21 +11,33 @@ module AutocompleteSource
   # from collection titles
   AUTOCOMPLETE_WORD_TERMINATOR = ",,".freeze
 
+  def transliterate(input)
+    input = input.to_s.mb_chars.normalize(:kd).gsub(/[\u0300-\u036F]/, "")
+    result = ""
+    input.each_char do |char|
+      tl = ActiveSupport::Inflector.transliterate(char)
+      # Don't transliterate characters which are not supported (e.g. non-Latin characters).
+      # No need for special handling of "?" since it is not modified by transliteration
+      result << (tl == "?" ? char : tl)
+    end
```

Comment:
I think this might be clearer if the code were less concise, particularly since the character we're dealing with is a question mark, which is used in the ternary operator. Maybe something like:

```ruby
if tl == "?"
  # If transliterate returns "?", the original character is either unsupported (e.g. a non-Latin character) or was actually a question mark. In both cases, we should keep the original.
  result << char
else
  result << tl
end
```